### PR TITLE
with changes

### PR DESCRIPTION
**Pull request by hypervisorfdi**
To resolve :

	//TODO this is anti pattern, we should use constructor to inject/inject class's fields , but @singelton dosent work , i dont know why
	// thats why i make this property/field as static so the DB/Map will hold the data during the application life time, other wise 
	//new map will be created every time we call this class

### CODE REVIEW

File: src/main/java/com/worldline/codetest/FatJarApplication.java [hunk 1]
```diff
@@ -17,23 +22,26 @@
  */
 public class FatJarApplication extends Application {
 
-    HashSet<Object> singletons = new HashSet<Object>();
+    private static final HashSet<Object> singletons = new HashSet<Object>();
+    private static final HashSet<Class<?>> classes = new HashSet<Class<?>>();
 
     public FatJarApplication() {
+        ProfileService profileService = new ProfileService(new ProfileRepositoryMapImp());
```
Comment (mutazkabashi) [inline src/main/java/com/worldline/codetest/FatJarApplication.java:29]: Nice Refactoring, I wasn't aware about the `getSingeltion`() method could be used to create Singelton classes

File: src/main/java/com/worldline/codetest/rest/ProfileApi.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
+package com.worldline.codetest.rest;
```
Comment (mutazkabashi) [inline src/main/java/com/worldline/codetest/rest/ProfileApi.java:1]: Nice Refactoring, Moving the Swagger Specification to an interface

**Review by mutazkabashi — APPROVED** [review]

File: src/main/java/com/worldline/codetest/FatJarApplication.java (hunk 1 above)
Comment (hypervisorfdi) [reply src/main/java/com/worldline/codetest/FatJarApplication.java:29]: What we can conclude here is that in JAX-RS

- Singletons are already constructed API:s

- Classes are hint to JAX-RS specification implementation ( RestEasy or other implementations) on what to be initiated with a default empty constructor ( requirement ).

take away lesson here is that dependency injection is for free in Java, aka Constructors.

File: src/main/java/com/worldline/codetest/rest/ProfileApi.java (hunk 2 above)
Comment (hypervisorfdi) [reply src/main/java/com/worldline/codetest/rest/ProfileApi.java:1]: JAX-RS annotations and swagger makes classes hard to read.

Im fan of interfaces, especially default methods in Interfaces ( Java8 ), wish more start using it.

**Pull request closed without merging**